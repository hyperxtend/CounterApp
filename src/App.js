import React, { Component } from 'react';

class App extends Component {
  constructor() {
    super();
    this.state = {
      count: 0,
    }
  }

  createIncrement = amount => () =>
    this.setState(prevState => ({
      count: prevState.count + amount,
    }));

  increases = this.createIncrement(1);

  decreases = this.createIncrement(-1);

  render() {
    return (
      <div>
        <p>Count: {this.state.count}</p>
        <button className="increase" onClick={this.increases}>+</button>
        <button className="decrease" onClick={this.decreases}>-</button>
      </div>
    )
  }
}

export default App;
