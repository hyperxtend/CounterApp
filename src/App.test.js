import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('<App/>', () => {
  it('checks initial count to be 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 0');
  });

  it('checks if count increases by 1', () => {
    const wrapper = shallow(<App />);
    const increase = wrapper.find('button.increase');
    increase.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 1');
  });

  it('checks if count increases by 1', () => {
    const wrapper = shallow(<App />);
    const decrease = wrapper.find('button.decrease');
    decrease.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: -1');
  });

  it('creates a snapshot of component', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
