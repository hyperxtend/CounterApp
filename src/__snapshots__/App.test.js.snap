// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> creates a snapshot of component 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    className="increase"
    onClick={[Function]}
  >
    +
  </button>
  <button
    className="decrease"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
